joplin/pages/official_documents_page/management/commands/extract_document_text.py

```diff
@@ -0,0 +1,53 @@
+import urllib3
+import io
+import fitz
+from django.core.management.base import BaseCommand, CommandError
+from pages.official_documents_page.models import OfficialDocumentPage
+
+
+def extract_text_from_url(url):
+    http = urllib3.PoolManager()
+    resp = http.request('GET', url)
+    file_stream = io.BytesIO(resp.data)
+    text = ''
+    try:
+        pdf_doc = fitz.open(stream=file_stream, filetype='pdf')
+        for page in pdf_doc:
+            text += page.getText()
+    except RuntimeError:
+        print(f'Runtime Error for {url}')
+    return text
+
+
+def extract_document_text():
+    all_document_pages = OfficialDocumentPage.objects.all()
+    print(f'Beginning extraction...')
+
+    url_base = 'https://joplin3-austin-gov-static.s3.amazonaws.com/production'
+
+    for page in all_document_pages:
+        '''
+        get the document url
+        get the data from the document url
+        set body as text and save
+        '''
+        print(f'Page id {page.id}')
+        # Check if page body already has content, if so skip
+        if len(page.body) > 0:
+            continue
+        if page.document and page.document.url:
+            filename = page.document.url.split('docs')[1]
+            print(f'reading {filename}')
+            doc_url = url_base + filename
+            extracted_text = extract_text_from_url(doc_url)
+            page.body = extracted_text
+            page.save()
```

Comment:
Do we want these pages to be saved or published? Or does it depend on what its current status is?

Reply:
Saving a page doesn't change its publish status one way or another. 
I just tested it, if a page is live, page.save() updates the live page
if the page is in draft, it stays in draft.